src/cargo/ops/lockfile.rs

```diff
@@ -97,6 +91,23 @@ pub fn write_pkg_lockfile(ws: &Workspace, resolve: &Resolve) -> CargoResult<()>
     })
 }
 
+fn are_equal_lockfiles(mut orig: String, current: &str, strict: bool) -> bool {
```

Comment:
I think `strict` here has an inverse meaning of its name, if `strict` is true that means that we can update the lock file, but I'd expect `strict: true` to mean that updates are disallowed? Perhaps the `Config` could be passed in here to be inspected locally where necessary?

Reply:
Yeah, naming is confusing indeed, I've flipped between `strict` and `relaxed` several times. In itself, I think `strict` is right though: strict comparison means that we do byte-for-byte comparison, and `!strict` means that we deem lockfiles equal even if there are minor differences. The tricky bit is that we need to do approximate comparison if lockfile update is not allowed :) 

I'll just rename this flag to `lock_update_allowed` to avoid inventing confusing abstractions :) Re passing `config`, I doubt we ever need anything except `lock_update_allowed` boolean, and in theory keeping this flag a simple bool allows us to write unit-tests for this function :)